### PR TITLE
TransposeSinking: StridedSlice op support

### PR DESCRIPTION
**Pull request by itikhono**
### Details:
TSStridedSliceForward and TSStridedSliceBackward transformations
Tests

### Tickets:
 - *105220*

### CODE REVIEW

File: src/common/transformations/src/transformations/transpose_sinking/ts_strided_slice.cpp [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+            return false;
+        }
+
+        auto data_partial_shape = strided_slice->input_value(0).get_partial_shape();
+        auto data_rank = data_partial_shape.rank();
+        if (data_rank.is_dynamic()) {
+            return false;
+        }
+
+        // todo: handle ellipsis_mask
```
Comment (itikhono) [inline src/common/transformations/src/transformations/transpose_sinking/ts_strided_slice.cpp:264]: create a ticket

File: src/common/transformations/src/transformations/transpose_sinking/ts_strided_slice.cpp [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+                return false;
+            }
+
+            auto input_const_val = input_const->cast_vector<uint64_t>();
+            if (input_idx == 1) {
+                // `begin` input have to be initialized with 0
+                input_const_val.resize(expected_size, 0);
+                num_elements_in_begin_input = num_elements;
+            } else if (input_idx == 2) {
+                // 'end' input have to be initialized with the corresponding `data` input dim value
```
Comment (itikhono) [inline src/common/transformations/src/transformations/transpose_sinking/ts_strided_slice.cpp:137]: fix the comment

File: src/common/transformations/include/transformations/transpose_sinking/ts_strided_slice.hpp [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+namespace pass {
+namespace transpose_sinking {
+
+class TRANSFORMATIONS_API TSStridedSliceForward;
+class TRANSFORMATIONS_API TSStridedSliceBackward;
+
+}  // namespace transpose_sinking
+}  // namespace pass
+}  // namespace ov
+
```
Comment (evkotov) [inline src/common/transformations/include/transformations/transpose_sinking/ts_strided_slice.hpp:22]: Could you please add some common comments about transformations.

File: src/common/transformations/src/transformations/transpose_sinking/ts_strided_slice.cpp (hunk 2 above)
Comment (itikhono) [reply src/common/transformations/src/transformations/transpose_sinking/ts_strided_slice.cpp]: done

File: src/common/transformations/src/transformations/transpose_sinking/ts_strided_slice.cpp (hunk 1 above)
Comment (itikhono) [reply src/common/transformations/src/transformations/transpose_sinking/ts_strided_slice.cpp:264]: done

File: src/common/transformations/tests/transpose_sinking/ts_strided_slice.cpp [hunk 4]
```diff
@@ -0,0 +1,666 @@
...
+INSTANTIATE_TEST_SUITE_P(TSCommonStridedSliceBackward_2, TSTestFixture, test_backward_strided_slice(bw_test_2()));
+INSTANTIATE_TEST_SUITE_P(TSCommonStridedSliceBackward_3, TSTestFixture, test_backward_strided_slice(bw_test_3()));
+INSTANTIATE_TEST_SUITE_P(TSCommonStridedSliceBackward_4, TSTestFixture, test_backward_strided_slice(bw_test_4()));
+INSTANTIATE_TEST_SUITE_P(TSCommonStridedSliceBackward_5, TSTestFixture, test_backward_strided_slice(bw_test_5()));
+INSTANTIATE_TEST_SUITE_P(TSCommonStridedSliceBackward_6, TSTestFixture, test_backward_strided_slice(bw_test_6()));
+INSTANTIATE_TEST_SUITE_P(TSCommonStridedSliceBackward_7, TSTestFixture, test_backward_strided_slice(bw_test_7()));
+INSTANTIATE_TEST_SUITE_P(TSCommonStridedSliceBackward_8, TSTestFixture, test_backward_strided_slice(bw_test_8()));
+INSTANTIATE_TEST_SUITE_P(TSCommonStridedSliceBackward_9, TSTestFixture, test_backward_strided_slice(bw_test_9()));
+
+}  // namespace gather
```
[the remark this answers is not in the document]
Comment (itikhono) [reply src/common/transformations/tests/transpose_sinking/ts_strided_slice.cpp]: fixed

File: src/common/transformations/include/transformations/transpose_sinking/ts_strided_slice.hpp (hunk 3 above)
Comment (itikhono) [reply src/common/transformations/include/transformations/transpose_sinking/ts_strided_slice.hpp:22]: done

**Review by evkotov — APPROVED** [review]

**Comment by itikhono** [comment]
This PR was postponed because it affects performance of TF models

**Pull request closed without merging**